unit_tests/kernels/UnitTestKernelUtils.C

```diff
@@ -382,6 +385,417 @@ void calc_mass_flow_rate_scs(
     });
 }
 
+void calc_projected_nodal_gradient_interior(
+  const stk::mesh::BulkData& bulk,
+  const stk::topology& topo,
+  const VectorFieldType& coordinates,
+  const ScalarFieldType& dnv,
+  const ScalarFieldType& scalarField,
+  const VectorFieldType& gradField)
+{
+  const auto& meta = bulk.mesh_meta_data();
+  const int ndim = meta.spatial_dimension();
+  EXPECT_EQ(ndim, 3);
+
+  sierra::nalu::ElemDataRequests dataNeeded;
+
+  auto meSCS = sierra::nalu::get_surface_master_element(topo);
+
+  dataNeeded.add_cvfem_surface_me(meSCS);
+  dataNeeded.add_coordinates_field(coordinates, ndim, sierra::nalu::CURRENT_COORDINATES);
+  dataNeeded.add_gathered_nodal_field(scalarField, 1);
+  dataNeeded.add_gathered_nodal_field(dnv, 1);
+  dataNeeded.add_gathered_nodal_field(gradField, ndim);
+  dataNeeded.add_master_element_call(sierra::nalu::SCS_AREAV, sierra::nalu::CURRENT_COORDINATES);
+
+  const stk::mesh::Selector selector = meta.locally_owned_part() | meta.globally_shared_part();
+  const auto& buckets = bulk.get_buckets(stk::topology::ELEM_RANK, selector);
+
+  const int bytes_per_team = 0;
+  const int bytes_per_thread = sierra::nalu::get_num_bytes_pre_req_data(dataNeeded, meta.spatial_dimension()) ;
+
+  auto v_shape_function = Kokkos::View<double**>("shape_function", meSCS->numIntPoints_, meSCS->nodesPerElement_);
+
+  auto team_exec = sierra::nalu::get_team_policy(buckets.size(), bytes_per_team, bytes_per_thread);
+
+  Kokkos::parallel_for(team_exec, [&](const sierra::nalu::TeamHandleType& team) {
```

Comment:
Can we not use `kokkos_thread_team_bucket_loop` from `UnitTestKokkosUtils.h` here? That will reduce some of the boilerplate code that we have here.